### PR TITLE
Add Curriculum Catalog Language Row

### PR DESCRIPTION
**Pull request by TurnerRiley**
Adds a row below the catalog filters when the user is not in an English locale. This row displays the total number of curricula translated in that language, and provides a toggle for the user to filter for said translated curricula.

### Row not appearing in English locales
English

### Row appearing in non-English locales
Spanish

### Lists correct number of translated curricula based on locale
French:
French_num
Arabic:
arabic_num
German:
German_num

### Allows the user to filter for translated curricula
https://github.com/code-dot-org/code-dot-org/assets/56283563/b4e72cad-a017-4eef-84b7-5b2058eb1d0d

## Links
Jira ticket: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?modal=detail&selectedIssue=ACQ-491&assignee=60d62161f65054006980bd71)
Spec: [here](https://docs.google.com/document/d/1UID-xzCF5gxOHihcYiYuZwpotCpGXGPTzqX1UyS37-I/edit#heading=h.qi8xxyq72bct)

## Testing story
Local testing and adding unit tests.

## Follow-up work
Currently, this is the only filter that does not read from or update the URL params (such that a user could leave this page and return with their filters still applied). That is a future ticket tracked [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?modal=detail&selectedIssue=ACQ-671&assignee=60d62161f65054006980bd71).

## PR Checklist:
- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by dmcavoy** [comment]
So excited to see this!!! Last major feature for Curriculum Catalog V1

File: apps/test/unit/templates/curriculumCatalog/curriculumCatalogCardTest.jsx [hunk 1]
```diff
@@ -184,16 +184,39 @@ describe('CurriculumCatalogCard', () => {
...
+      ...defaultProps,
+      isEnglish: false,
+      isTranslated: true,
+    });
+
+    expect(screen.queryByTitle(translationIconTitle)).to.be.null;
+    expect(container.querySelectorAll('i[class*=language]')).to.have.length(0);
+  });
+
+  it('renders translation icon when translation is available not in English locale', () => {
```
Comment (davidsbailey) [inline apps/test/unit/templates/curriculumCatalog/curriculumCatalogCardTest.jsx]: remove duplicate test case?

**Review by davidsbailey — APPROVED** [review]
the final piece 🧩 !

File: apps/test/unit/templates/curriculumCatalog/curriculumCatalogCardTest.jsx (hunk 1 above)
Comment (TurnerRiley) [reply apps/test/unit/templates/curriculumCatalog/curriculumCatalogCardTest.jsx]: Whoops thanks for spotting that! Hoisted by my copy/paste-ing

File: apps/src/templates/curriculumCatalog/CurriculumCatalog.jsx [hunk 2]
```diff
@@ -320,6 +335,32 @@ const CurriculumCatalog = ({curriculaData, ...props}) => {
           color={Button.ButtonColor.brandSecondaryDefault}
         />
       </div>
+      {!isEnglish && (
+        <div className={style.catalogLanguageFilterRow}>
+          <div className={style.catalogLanguageFilterRowNumAvailable}>
+            <BodyOneText>
+              {i18n.numCurriculaAvailableInLanguage({
```
Comment (elf-code) [inline apps/src/templates/curriculumCatalog/CurriculumCatalog.jsx:342]: I came here to check that this was going to get translated eventually, thank you! 😺

**Pull request merged**